### ISSUE TITLE
Option to enforce max one tag per file for namespaces with mutex tags

### ISSUE THREAD

**Issue by novakda** [hydrusnetwork/hydrus#404]
Some namespaces are used for mutually elusive values (you wouldn't want more than one of these tags on the same file, except in rare cases)

It would be nice to have some way to enforce or validate this while adding tags, possibly with a way to turn this mode on or off on the fly.

For example, there could be a list of namespaces under options, such that when you attempt to add a tag from any of these namespaces and the file already has one, the system will:

- just prevent you from doing it
- or, display a confirmation dialog (with options to add the new tag, replace the existing one in the same namespace or cancel)

**Comment by Zweibach** [comment]
Kindly provide examples.

**Comment by novakda** [comment]
> Kindly provide examples.

I use a "motive" namespace to describe what format the original version was created in:

- motive:artwork
- motive:photo
- motive:video
- motive:animation
- motive:document
- motive:thumbnail
- motive:music
- motive:oil painting
- motive:sculpture

.. and so on.

Generally, I wan't only one of these per file, so I can detect where they are missing.   

I have other cases where more than one is possible, but uncommon, such as "author:" or "creator:".  For these I'd want some kind of confirmation so it doesn't go unchecked.

Another example would be ISBN numbers for books, or ID numbers from stock photography sites.

**Comment by Mengmoshu** [comment]
I do something very similar, except I called my namespace `media:`. So what would you do if an image contained multiple independent works in different mediums? I just slap all applicable tags on, regardless of the originality involved. So a reference photo in the same image as the drawing produced will have both `media:photo` and `media:drawn`.

Though I do support this feature request, it's the sort of thing that should be used with care. And might even need to be locked behind janitor accounts for the PTR and other tag repo servers.

Also, since this example has a clear flaw I've seen in the wild we might want to look for another example. Which might be difficult given that the power of tags is flexible non-exclusive non-hierarchical normalized (mostly) metadata.

**Comment by DonaldTsang** [comment]
Agreed, enforcing rigidity of intention, or category is more of a user preference than anything else, it could be there but that can be handled later.
However, standard file or media IDs is absolutely useful... until you realize that the same ID can have different image resolutions (see Tumblr's 500 vs 1280).
